### PR TITLE
Crm457 1458 assessment comments

### PR DESCRIPTION
**Pull request by ivanELEC**
## Description of change

 [CRM7: Pull In Assessment Comments on CRM7 Claims](https://dsdmoj.atlassian.net/browse/CRM457-1458)

### Before changes:
image

### After changes:
image

### CODE REVIEW

File: app/presenters/nsm/check_answers/application_status_card.rb [hunk 1]
```diff
@@ -122,12 +122,13 @@ def update_claim
...
       def response
-        @response ||= begin
-          response_text = "Fake LAA Response\nSecond line"
-          status.submitted? ? [] : response_text.split("\n").map { sanitize(_1) }
-        end
+        @response ||= if status.granted?
+                        [I18n.t('nsm.steps.view_claim.granted_response')]
+                      else
+                        response_text = claim.assessment_comment
+                        status.submitted? ? [] : response_text.split("\n").map { sanitize(_1) }
```
Comment (jsugarman) [inline app/presenters/nsm/check_answers/application_status_card.rb:130]: you may be able to use [simple_format](https://crm457-1381-amend-nscc-provider-dev.cloud-platform.service.justice.gov.uk/) here and not decontruct response on calling (at line 32)
```suggestion
                        status.submitted? ? '' : simple_format(response_text)
```

File: app/services/app_store_update_processor.rb [hunk 2]
```diff
@@ -18,10 +19,13 @@ def convert_params(record)
...
 
-    def update_claim(claim_id, params)
+    def update_claim(claim_id, params, record)
       claim = Claim.find_by(id: claim_id)
 
-      claim&.update!(params)
+      return unless claim
+
+      claim.update!(params)
+      Nsm::AssessmentSyncer.call(claim, record:)
```
Comment (jsugarman) [inline app/services/app_store_update_processor.rb:28]: possible refactor

```ruby
    def update_claim(record)
      claim = Claim.find_by(id: record['application_id'])
      
      return unless claim

      claim.update!(convert_params(record))
      Nsm::AssessmentSyncer.call(claim, record:)
    end
```

and call at line 6

```ruby
  update_claim(record
```

and similar for `update_prior_authority_application` method

File: spec/services/submit_to_app_store/nsm_payload_builder_spec.rb [hunk 3]
```diff
@@ -15,7 +16,8 @@
     travel_to(Time.zone.local(2023, 8, 17, 12, 13, 14)) do
       check_json(subject.payload).matches(
         application: {
-          'agent_instructed' => 'no',
+          'assessment_comment' => 'this is an assessment',
+        'agent_instructed' => 'no',
```
Comment (jsugarman) [inline spec/services/submit_to_app_store/nsm_payload_builder_spec.rb:20]: indent missing?

**Review by jsugarman — APPROVED** [review]
LGTM - couple of possible refactors suggested

File: app/presenters/nsm/check_answers/application_status_card.rb (hunk 1 above)
Comment (ivanELEC) [reply app/presenters/nsm/check_answers/application_status_card.rb:130]: using simple_format messes with how the text is combined in the join_strings method so going to leave this as it is for now

File: app/services/app_store_update_processor.rb [hunk 4]
```diff
@@ -18,10 +18,13 @@ def convert_params(record)
...
+      claim = Claim.find_by(id: record['application_id'])
 
-      claim&.update!(params)
+      return unless claim
+
+      claim.update!(convert_params(record))
+      Nsm::AssessmentSyncer.call(claim, record:)
     end
 
     def update_prior_authority_application(application_id, params, record)
```
Comment (jsugarman) [inline app/services/app_store_update_processor.rb:30]: @ivanELEC for consistency of interfaces it feels like this method should be changed to do the same as `update_claim`. But I have already approved so leave it up to you

Comment (ivanELEC) [reply app/services/app_store_update_processor.rb]: good shout - just added this in

**Pull request merged**